src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1800,7 +1800,7 @@ public static Tombstone TableValueToTombstone(JsonOperationContext context, ref
                     var shouldVersion = DocumentDatabase.DocumentsStorage.RevisionsStorage.ShouldVersionDocument(
                         collectionName, nonPersistentFlags, local.Document.Data, null, context, id, lastModifiedTicks, ref flags, out var configuration);
 
-                    if (shouldVersion)
+                    if (shouldVersion || flags.Contain(DocumentFlags.HasRevisions))
```

Comment:
if I did force created revision, and after that modify the document again (without force created).
This will not cause for a new revision to be created, right?

Reply:
okay, so you create a `DeleteRevision` if needed, to avoid a situation that you have revisions to the document that we are about to delete, but `shouldVersion` is false, so you will end up with orphaned revisions.
add a comment about this case.

Reply:
That's what it does actually..It doesnt touch the revisions because the document has no relevant revisions config..
*BUT* it still create the `Delete Revision`..
I think it is better this way because it is going through one code path..